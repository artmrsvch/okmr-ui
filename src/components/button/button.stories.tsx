import * as React from 'react';
import { Story } from '@storybook/react/types-6-0';
import 'antd/lib/button/style/index.css';
import Button from './index';

import {
  Title,
  Subtitle,
  Description,
  Primary,
  ArgsTable,
  Stories,
  PRIMARY_STORY
} from '@storybook/addon-docs/blocks';
import { xs } from '../../config/breakpoints';

export default {
  title: 'Components/Button',
  component: Button,
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Subtitle />
          <Description>
            {`Стандартная кнопка из библиотека antd, которая отстилизована согласно макету. Помимо
                  стандартных props, появился "color", который согласно цветовой схеме макета добавляет нужный цвет.
                  На разрешении экрана ${xs} и ниже, кнопка занимает всю ширину экрана "width: 100%"`}
          </Description>
          <Primary />
          <ArgsTable story={PRIMARY_STORY} />
          <Stories />
        </>
      )
    }
  }
};

const Template: Story = (args) => <Button {...args} />;

export const Default = Template.bind({});

Default.args = {
  type: 'primary',
  danger: false,
  children: 'Кнопка',
  loading: false
};
Default.argTypes = {
  type: {
    description: 'Типы кнопок',
    control: {
      type: 'select',
      options: ['primary', 'link', 'default', 'dashed', 'text']
    }
  },
  color: {
    description: 'Кастомные цвета из макета',
    control: {
      options: ['blue', 'green', 'orange', 'lightBlue'],
      type: 'select'
    }
  }
};
