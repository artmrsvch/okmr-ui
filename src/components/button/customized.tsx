import * as React from 'react';
import { ButtonProps } from 'antd';
import { Button } from '../index';

export interface CustomizedButtonProps {
  suffix?: React.ReactNode | string;
  prefix?: React.ReactNode | string;
  children?: string | React.ReactNode;
  className?: string;
}

export type CustomizedProps = CustomizedButtonProps & ButtonProps;

const CustomizedButton: React.FC<CustomizedProps> = ({
  suffix,
  prefix,
  children,
  ...restProps
}) => (
  <Button icon={prefix} {...restProps}>
    {children || null}
    {suffix}
  </Button>
);

export default CustomizedButton;
