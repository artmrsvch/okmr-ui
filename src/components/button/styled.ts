import styled from 'styled-components';
import { Button, ButtonProps } from 'antd';
import { Colors } from 'src/config/colors';
import { xs } from 'src/config/breakpoints';

interface CustomStyledButton extends ButtonProps {
  color?: 'orange' | 'green' | 'red' | 'lightBlue' | 'yellow' | 'blue';
}

export const StyledAtndButton = styled(Button)`
  &,
  &:hover,
  &:active,
  &:focus {
    background: ${(pr) =>
      pr.type !== 'link' &&
      pr.type !== 'dashed' &&
      pr.type !== 'default' &&
      pr.type !== 'text' &&
      pr.color === 'red'
        ? '#FF6960'
        : pr.type !== 'link' &&
          pr.type !== 'dashed' &&
          pr.type !== 'default' &&
          pr.type !== 'text' &&
          pr.color === 'yellow'
        ? '#fff566'
        : pr.type !== 'link' &&
          pr.type !== 'dashed' &&
          pr.type !== 'default' &&
          pr.type !== 'text' &&
          pr.color === 'blue'
        ? '#40A9FF'
        : pr.type !== 'link' &&
          pr.type !== 'dashed' &&
          pr.type !== 'default' &&
          pr.type !== 'text' &&
          pr.color === 'green'
        ? '#A8CD3D'
        : pr.type !== 'link' &&
          pr.type !== 'dashed' &&
          pr.type !== 'default' &&
          pr.type !== 'text' &&
          pr.color === 'orange'
        ? '#FF9B39'
        : undefined};
    border-color: ${(pr: CustomStyledButton) =>
      pr.color && pr.type !== 'link' && pr.type !== 'text' ? `${Colors[pr.color]}` : undefined};
    color: ${(pr: CustomStyledButton) =>
      pr.color && pr.type !== 'primary' ? `${Colors[pr.color]}` : undefined};
  }

  @media screen and (max-width: ${xs}) {
    width: 100%;
  }
`;
