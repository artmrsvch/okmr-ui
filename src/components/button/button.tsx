import {StyledAtndButton as Button} from "./styled";
import 'antd/lib/button/style/css'

export default Button
