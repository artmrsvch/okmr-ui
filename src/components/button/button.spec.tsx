import React from 'react';

import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';

// üëá Imports a specific story for the test
import { CutomizedButtonWithIcon } from './button.stories';

it('renders the button in the primary state', () => {
  render(<CutomizedButtonWithIcon>–û—Ç–∫—Ä—ã—Ç—å</CutomizedButtonWithIcon>);
  expect(screen.getByRole('button')).toHaveTextContent('–û—Ç–∫—Ä—ã—Ç—å');
});
