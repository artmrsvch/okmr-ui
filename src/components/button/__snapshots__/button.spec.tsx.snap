// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Test renders a disabled button with prop disabled 1`] = `
<Button
  className="sc-bdnxRM fIvcYL"
  disabled={true}
>
  Button Test
</Button>
`;

exports[`Button Component Test renders button with a loading icon when prop loading is true 1`] = `
<Button
  className="sc-bdnxRM fIvcYL"
  loading={true}
>
  Button Test
</Button>
`;

exports[`Button Component Test renders ghost style with prop ghost 1`] = `
<Button
  className="sc-bdnxRM fIvcYL"
  ghost={true}
>
  Button Test
</Button>
`;

exports[`Button Component Test renders without child element 1`] = `
<Button
  className="sc-bdnxRM fIvcYL"
>
  Button Test
</Button>
`;

exports[`Button Component Test simulates a click event 1`] = `
<Button
  className="sc-bdnxRM fIvcYL"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Button Test
</Button>
`;

exports[`Button matches snapshot: invisible 1`] = `ShallowWrapper {}`;
